src/System.Drawing.Common/src/System/Drawing/Printing/PreviewPrintController.Unix.cs

```diff
@@ -32,40 +32,31 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
-using System;
-using System.Collections;
-using System.Drawing.Imaging;
-
 namespace System.Drawing.Printing
 {
-    public class PreviewPrintController : PrintController
+    public partial class PreviewPrintController : PrintController
     {
-        bool useantialias;
-        ArrayList pageInfoList;
-
-        public PreviewPrintController()
-        {
-            pageInfoList = new ArrayList();
-        }
-        public override bool IsPreview
-        {
-            get { return true; }
-        }
-
         public override void OnEndPage(PrintDocument document, PrintPageEventArgs e)
         {
         }
 
         public override void OnStartPrint(PrintDocument document, PrintEventArgs e)
         {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
```

Comment:
Why is this necessary?  Internal callers of this won't pass null, and this shouldn't be called by external callers (even though technically it could be).  Why is it better to add code and cost rather than just letting it null ref when it's used in a way it shouldn't be used?

Reply:
+1, outside of public API this kind of check is probably only worthwhile in cases where it will not null ref til later (eg., on a constructor when the null ref occurs on a method later)

Reply:
@hughbe planning on addressing these comments?